### DISCUSSION TITLE
Treat data before export to xls

### DISCUSSION THREAD

**Discussion by Honokai** [Power-Components/livewire-powergrid#246] [Ideas]
Is it possible to treat data before exporting? I am currently unable to export to xls because it throws error InvalidArgumentException Null, I know that it is an Object and not null as it says, I would like to know if I could treat it before actually dispatching to queue.

**Comment by dansysanalyst** [comment]
Hi Honokai,

Maybe this issue can help you:

https://github.com/Power-Components/livewire-powergrid/issues/176

Thanks for downloading!

**Reply by Honokai** [reply]
Unfortunately it's not the same, let me show you somewhat what it looks like.

I have a powergrid table, let's call it **InvoiceTable** and it's model is Invoice and it's attributes are:

array({ 
  client_id: array([...client_attributes]), // returns a collection of Client::class, meaning there is a relation here between tables
  purchase_value: float,
  date_purchase: new Date(),
  transaction_pass: uuid,
})


So my InvoiceTable will show some of the attributes above, let's say i want to show only client and purchase_value, i would do:

```
return PowerGrid::eloquent()
 ...

Collumn::add('client', function (Client $model) {
    return $model->client->name;
})
->Collumn::add('purchase_value')

```


```
public function columns(): array
    {
        return [
           ....
           Column::add()
                ->title(__('Client name'))
                ->field('client', 'clients.name')
                ->searchable()
                ->makeInputText(),

            Column::add()
                ->title(__('Purchase value'))
                ->field('purchase_value')
                ->searchable()
                ->makeInputText(),

          ];
```


When I try to export I get an error, which is the one I'm trying to solve, basically, what the export function is accessing isn't any of the supported types(string, boolean), but its an Array that contains the object Client in this case, it should be accessing the attribute **name**, but it receives Object instead.

dd() before error gives something like this:


    id: 1
    client: array() //it's when trying to export this data that it throws the error
    purchase_value: 1200
    transaction_pass: 1e2w523561
